### PR TITLE
Add single commit check

### PR DESCRIPTION
**Pull request by mneumark**
[Card](https://rentpath.atlassian.net/browse/SRV-5156)

* create commit-count-check workflow
* implement commit-count-check workflow inside elixir-build-check

### CODE REVIEW

File: .github/workflows/elixir-build-check.yaml [hunk 1]
```diff
@@ -114,3 +114,8 @@ jobs:
           git diff --name-only "$default_branch_ref" | \
           egrep '(\.ex|\.exs)+$' | \
           xargs --no-run-if-empty mix format --check-formatted --dry-run
+  single-commit-check:
```
Comment (Stratus3D) [inline .github/workflows/elixir-build-check.yaml:117]: I'm not sure this really belongs in the `elixir-build-checks` workflow. It's not Elixir specific.

Comment (mneumark) [reply .github/workflows/elixir-build-check.yaml:117]: See the discussion in #elixir-dev in slack

File: .github/workflows/elixir-build-check.yaml [hunk 2]
```diff
@@ -114,3 +114,8 @@ jobs:
...
+    name: Check For Single Commit
+    uses: ./.github/workflows/commit-count-check.yaml
```
Comment (KevinMarth) [inline .github/workflows/elixir-build-check.yaml:119]: Nitpick: the leading `./` doesn't add (or prevent) anything here.

Comment (mneumark) [reply .github/workflows/elixir-build-check.yaml:119]: I tried wiithout it and it wouldn't work.

Comment (KevinMarth) [reply .github/workflows/elixir-build-check.yaml:119]: OK, that is interesting. 🤷‍♂️

**Comment by mneumark** [comment]
Switching strategies to only enable squash merges. Closing

**Pull request closed without merging**